lib/publiccloud/instance.pm

```diff
@@ -264,6 +264,36 @@ sub upload_log {
     assert_script_run("test -d '$tmpdir' && rm -rf '$tmpdir'");
 }
 
+=head2 verify_upload_logs_from_dir
+
+    verify_upload_logs_from_dir($dir, @files, $log_header);
+
+Check files status and upload logs ok to UI. This works on files in a specific directory.
+Input: C<$dir> directory path, C<@files> files basenames array; C<$log_header> the leading name added to each log
+
+For more directories to manage, repeated calls for each one can be done.
+Still, you can also set empty dir "" and use full path names, from root.
+=cut
+
+sub verify_upload_logs_from_dir {
+    my ($self, $dir, @files, $log_header) = @_;
+    my @logs = ();
+    my @missing_logs = ();
+
+    # Select existing files
+    foreach (@files) {
+        my $res = $self->ssh_script_run(cmd => "test -e $dir/" . $_, quiet => 1, proceed_on_failure => 1);
+        $res = isok($res) ? push(@logs, $_) : push(@missing_logs, $_);
```

Comment:
Is `isok` from the `Test::More` lib?

Reply:
Isn't `$res` a simple scalar?

Reply:
`res` is a scalar and infact the value it receives from push is a count, the number of elements pushed, that in this case shall be 1 in any case: I could have checked it when not=1 for evental push issues, but avoided that more code.

`isok` is defined internal to that same module.

Reply:
ok, this line is a bit confusing though. I would rather see `if-else` here, as IMHO ternary is misused. I would expect that you wanna assign something to `$res` but any push should be successful unless the array exits.

Reply:
Where is `$res` used here after the ternary operator assignment?